### PR TITLE
IDEMPIERE-6147: Add IsHtml on AD_Field and AD_UserDef_Field

### PR DESCRIPTION
**Pull request by nmicoud**
https://idempiere.atlassian.net/browse/IDEMPIERE-6147

# Pull Request Checklist

- [X] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [X] My code follows the best practices of this project
- [X] I have performed a self-review of my own code
- [X] My code is easy to understand and review. 
- [X] I have checked my code and corrected any misspellings
- [X] In hard-to-understand areas, I have commented my code.
- [X] My changes generate no new warnings
### Tests
- [X] I have tested the direct scenario that my code is solving
- [X] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: migration/iD12/oracle/202405171607_IDEMPIERE-6147.sql [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
..., fg.fieldgrouptype AS FieldGroupType, fg.iscollapsedbydefault AS IsCollapsedByDefault, COALESCE(f.infofactoryclass, c.infofactoryclass) AS InfoFactoryClass, c.isautocomplete AS IsAutocomplete, COALESCE(f.isallowcopy, c.isallowcopy) AS IsAllowCopy, f.isdisplayedgrid AS IsDisplayedGrid, f.seqnogrid AS SeqNoGrid, c.seqnoselection AS SeqNoSelection, f.xposition AS XPosition, f.columnspan AS ColumnSpan, f.numlines AS NumLines, COALESCE(f.istoolbarbutton, c.istoolbarbutton) AS IsToolbarButton, c.formatpattern AS FormatPattern, f.isadvancedfield AS IsAdvancedField, f.isdefaultfocus AS IsDefaultFocus, COALESCE(f.ad_chart_id, c.ad_chart_id) AS AD_Chart_ID, f.ad_labelstyle_id AS AD_LabelStyle_ID, f.ad_fieldstyle_id AS AD_FieldStyle_ID, c.pa_dashboardcontent_id AS PA_DashboardContent_ID, COALESCE(trl.placeholder, ct.placeholder) AS Placeholder, COALESCE(f.ishtml, c.ishtml) AS IsHtml, f.isquickform AS IsQuickForm, COALESCE(f.ad_val_rule_lookup_id, c.ad_val_rule_lookup_id) AS AD_Val_Rule_Lookup_ID, vrf.code AS validationcodelookup, COALESCE(f.alwaysupdatablelogic, c.alwaysupdatablelogic) AS alwaysupdatablelogic, c.ad_infowindow_id AS AD_InfoWindow_ID, f.entitytype AS entitytype FROM ad_field f
```
Comment (CarlosRuiz-globalqss) [inline migration/iD12/oracle/202405171607_IDEMPIERE-6147.sql:143]: lines > 2499 characters are not valid in oracle sqlplus - this line needs to be split, and same for ad_field_v

File: org.adempiere.base/src/org/compiere/model/I_AD_Field.java [hunk 2]
```diff
@@ -75,6 +75,28 @@ public interface I_AD_Field
 
 	public org.compiere.model.I_AD_Column getAD_Column() throws RuntimeException;
 
+    /** Column name AD_Field_ID */
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.base/src/org/compiere/model/I_AD_Field.java:78]: is it possible that you are using a slightly different version of the core ModelGenerator classes?

why the order of the fields are different when you run it?

or is maybe the collation setting on your postgres?

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]

File: org.adempiere.base/src/org/compiere/model/I_AD_Field.java (hunk 2 above)
Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/compiere/model/I_AD_Field.java:78]: @nmicoud - what is the result of running the following query in your database?
```sql
SELECT 'AD_Field_UU' UNION
SELECT 'AD_Field_ID' UNION 
SELECT 'AD_Column_ID' UNION
SELECT 'AD_FieldGroup_ID'
ORDER BY 1
```

in my database the fields are ordered this way:
AD_Column_ID
AD_FieldGroup_ID
AD_Field_ID
AD_Field_UU

Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/compiere/model/I_AD_Field.java:78]: nevermind - I found the issue and opened ticket and pull request:
https://idempiere.atlassian.net/browse/IDEMPIERE-6149

**Comment by nmicoud** [comment]
BTW, there was a not commited change (moving scripts to iD11), as I think it can be integrated in stable version directly?

**Comment by nmicoud** [comment]
> > BTW, there was a not commited change (moving scripts to iD11), as I think it can be integrated in stable version directly?
> 
> Hi @nmicoud - better to keep it in iD12, it will be difficult to integrate because it conflicts with IDEMPIERE-3989

Done !

**Comment by CarlosRuiz-globalqss** [comment]
Thanks @nmicoud - sorry to bother you, can you please generate again the model classes after applying the fix for IDEMPIERE-6149 ?

**Comment by nmicoud** [comment]
> Thanks @nmicoud - sorry to bother you, can you please generate again the model classes after applying the fix for IDEMPIERE-6149 ?

Hi @CarlosRuiz-globalqss : done

**Review by CarlosRuiz-globalqss — APPROVED** [review]

**Pull request merged**